exotica_core/src/scene.cpp

```diff
@@ -612,7 +613,19 @@ void Scene::UpdateSceneFrames()
                 shape_transform.linear() = object.second->shape_poses_[i].rotation();
                 Eigen::Isometry3d trans = obj_transform.inverse() * shape_transform;
                 VisualElement visual;
-                visual.name = object.first;
+                std::string name = object.first;
+                // Check for name duplicates after loading from scene files
+                const auto& it = visual_map.find(name);
+                if (it != visual_map.end())
+                {
+                    it->second++;
+                    name = name + std::to_string(it->second);
```

Comment:
This makes it `Shape1` not `Shape_1` as in the PR description. Possibly worth printing a warning as well.

Reply:
Fixed.
Adding warnings will print them every time objects get added/removed. It would print warnings for calls that didn't cause the issue. I'd rather keep it without a warning.

Reply:
Oh, now I understand - so this is equivalent to `Shelf_collision_{n}`? Then perhaps add `_visual_` as a substring for consistency?

Reply:
The collision shapes do all get `_collision_#` prefix. For visuals, only the ones that got duplicated names (loaded from a scene file with multiple shapes defined). The naming is: `Shelf`, `Shelf_0`, `Shelf_1` ... when there are multiple meshes and correctly `Shelf` when there is only one. I'd rather keep it simple for now.

Reply:
Ok, I thought `_visual_` adds the same consistency as `_collision_` does.